cores/rp2040/SerialUSB.cpp

```diff
@@ -176,6 +177,15 @@ static bool _rts = false;
 static int _bps = 115200;
 static void CheckSerialReset() {
     if ((_bps == 1200) && (!_dtr)) {
+        // execute any pending tasks, so the host is happy
+        for(int i=0; i<1000; i++) tud_task();
+        // disable nvic irq, so that we don't get bothered anymore
+        NVIC_DisableIRQ((IRQn_Type) USBCTRL_IRQ);
+        // reset the whole USB hardware block
+        reset_block(RESETS_RESET_USBCTRL_BITS);
+        unreset_block(RESETS_RESET_USBCTRL_BITS);
+        // delay a bit...
+        for(int i=0;i<100000;i++);
```

Comment:
Well, mostly because I tend to not use SDK/RTOS, I'm unaware of what is available. 
Normally I write a BSP for Arduino bare-metal, including libraries. 

Depending on the optimization level, it certainly could be removed by the optimizer.
Since something better is available, and OK to call in the current context, then, fine. That's why I indicated there's room for improvement using a HW timer, CPU clock cycle counting, or something. :-)
I'll push the one-line fix shortly.

Reply:
No worries, it should be safe to `sleep_us/ms` here with the rest of the core/clocks trunning.

While doing that, could you also please replace the NVIC call with the SDK equivalent: `irq_set_enabled(USBCTRL_IRQ, false);` ?  We don't use the NVIC stuff directly elsewhere in the core.

Reply:
I'll do that, and also supply a guard that fixes another possible issue-- reentry :-)
I have some time finally to do this, and it's only a couple of lines.